src/ChunkMap.cpp

```diff
@@ -1711,97 +1711,107 @@ void cChunkMap::DoExplosionAt(double a_ExplosionSize, double a_BlockX, double a_
 			return;
 		}
 
-		for (int x = -ExplosionSizeInt; x < ExplosionSizeInt; x++)
+		for (int radius = 1; radius <= ExplosionSizeInt; ++radius)
 		{
-			for (int y = -ExplosionSizeInt; y < ExplosionSizeInt; y++)
+			for (int x = -radius; x <= radius; x++)
 			{
-				if ((by + y >= cChunkDef::Height) || (by + y < 0))
+				for (int y = -radius; y <= radius; y++)
 				{
-					// Outside of the world
-					continue;
-				}
-				for (int z = -ExplosionSizeInt; z < ExplosionSizeInt; z++)
-				{
-					if ((x * x + y * y + z * z) > ExplosionSizeSq)
+					if ((by + y >= cChunkDef::Height) || (by + y < 0))
 					{
-						// Too far away
+						// Outside of the world
 						continue;
 					}
-
-					BLOCKTYPE Block = area.GetBlockType(bx + x, by + y, bz + z);
-					switch (Block)
+					for (int z = -radius; z <= radius; z++)
 					{
-						case E_BLOCK_TNT:
+						if ((x * x + y * y + z * z) > ExplosionSizeSq)
 						{
-							// Activate the TNT, with a random fuse between 10 to 30 game ticks
-							int FuseTime = 10 + m_World->GetTickRandomNumber(20);
-							m_World->SpawnPrimedTNT(a_BlockX + x + 0.5, a_BlockY + y + 0.5, a_BlockZ + z + 0.5, FuseTime);
-							area.SetBlockTypeMeta(bx + x, by + y, bz + z, E_BLOCK_AIR, 0);
-							a_BlocksAffected.push_back(Vector3i(bx + x, by + y, bz + z));
-							break;
+							// Too far away
+							continue;
 						}
-
-						case E_BLOCK_OBSIDIAN:
-						case E_BLOCK_BEACON:
-						case E_BLOCK_BEDROCK:
-						case E_BLOCK_BARRIER:
-						case E_BLOCK_WATER:
-						case E_BLOCK_LAVA:
+						if ((abs(x) != radius) && (abs(y) != radius) && (abs(z) != radius))
 						{
-							// These blocks are not affected by explosions
-							break;
+							continue;
 						}
 
-						case E_BLOCK_STATIONARY_WATER:
+						BLOCKTYPE Block = area.GetBlockType(bx + x, by + y, bz + z);
+						switch (Block)
 						{
-							// Turn into simulated water:
-							area.SetBlockType(bx + x, by + y, bz + z, E_BLOCK_WATER);
-							break;
-						}
+							case E_BLOCK_TNT:
+							{
+								if (m_World->GetExplosionLimiter().RequestExplosion())
```

Comment:
>is there a chance that mine won't go off?

The limit does not cover manually ignited TNTs, so your first TNT would go off, but the remaining 4 TNTs have a chance of not exploding in a chain reaction.

Per chunk limit, rather than per-world, would solve this.